src/squashfs.hxx

```diff
@@ -266,9 +276,11 @@ public:
 class MetadataReader
 {
 	MetadataBlockReader f;
-        
-	static const size_t buf_size = 16 * squashfs::metadata_size;
-	char buf[buf_size];
+
+	static constexpr size_t max_supported_file_size = 1ull << 40; // 1 TiB -> 1 GiB buffer (2^30)
```

Comment:
Note, this doesn't take up 1 GiB of physical memory unless that part of the buffer is actually written to.
https://stackoverflow.com/questions/864416/are-some-allocators-lazy

Reply:
Wrote this comment in the code.

Reply:
Could you please clarify what the comment `// 1 TiB -> 1 GiB buffer (2^30)` means? Not sure I understand what you mean with it?

Reply:
The max file size we're declaring we support is 2^40 (1 TiB), which results in a max inode size of 2^30 (1 GiB) plus a few bytes. That means our buffer needs to be at least that large to read the inode.

1 TiB is arbitrary, but seemed future-proof since that's the size of our entire SSD.

Reply:
Can you please clarify the comment in code

Reply:
Done!